aws/rust-runtime/aws-config/src/profile/timeout_config.rs

```diff
@@ -117,10 +117,16 @@ impl ProfileFileTimeoutConfigProvider {
         let selected_profile = match profile.get_profile(selected_profile) {
             Some(profile) => profile,
             None => {
-                tracing::warn!(
-                    "failed to get selected '{}' profile, skipping it",
-                    selected_profile
-                );
+                // Lambdas don't have home directories and emitting this warning is not helpful
+                // to users running the SDK from within a Lambda. This warning will be silenced
+                // if we determine that that is the case.
+                let is_likely_running_on_a_lambda = check_is_likely_running_on_a_lambda(&self.env);
+                if !is_likely_running_on_a_lambda {
```

Comment:
This should behave the same as in `retry_config.rs` where it checks if the `profile_override` is set.